File: py/1011.py
from math import sqrt
import sys

read = sys.stdin.readline

num_of_cases = int(read())
for _ in range(num_of_cases):
  curr, dest = [int(x) for x in read().split()]
  distance = dest - curr
  n = int(sqrt(distance))
  count = n * 2
  distance -= n * (n + 1)
  count += distance // n
  count += 0 if distance % n == 0 else 1
  print(count)

File: java/1011.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {

	public static void main(String[] args) throws IOException {
		
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		
		int T = Integer.parseInt(br.readLine());
		int x = 0;
		int y = 0;
		
		for ( int i = 0 ; i < T ; i++ ) {
			
			StringTokenizer tz = new StringTokenizer(br.readLine()," ");
			
			x = Integer.parseInt(tz.nextToken());
			y = Integer.parseInt(tz.nextToken());
			
			System.out.println(cal(x, y));
			
		}

	}

	public static int cal(int x, int y) {
		
		int dis = y - x;
		int count = (int)Math.sqrt(dis);
		int res = 0;
	
		if( dis == count * count) {
			res = (2 * count) - 1;
		} else if ( dis <= count * (count + 1)  ) {
			res = (2 * count) ;
		} else {
			res = (2 * count) + 1;
		}
			
		return res;
		
	}

}
